src/context/index.tsx

```diff
@@ -86,3 +128,30 @@ export const withGlobalStyle = (Component: StyledSubcomponentType) => {
     return <ComponentWithGlobalStyles globalStyles={globalStyles} {...props} ref={ref} />;
   });
 };
+
+export const useAnalytics = () => {
+  const context = useContext(FoundryContext);
+  return (
+    componentType: string,
+    eventHandler: any,
+    eventType: string,
+    event: any,
+    props: any,
+  ): void => {
+    eventHandler(event);
+    const dateTime = new Date();
+    const deviceInfo: Record<string, unknown> = rdd.deviceDetect();
+    deviceInfo.innerHeight = window.innerHeight;
+    deviceInfo.innerWidth = window.innerWidth;
+
+    context.analyticsFunction(
+      componentType,
+      eventType,
+      event,
+      dateTime,
+      deviceInfo,
+      window.location.href,
```

Comment:
@dawsonbooth does accessing window like this make problems for SSR?

Reply:
Hmm, that looks to be the case:
- https://github.com/gatsbyjs/gatsby/issues/19294
- https://stackoverflow.com/a/66106510
SSR frameworks like Gatsby or Next will provide something like a `location` prop that can substitute for the current URL.

Reply:
Next.js: https://stackoverflow.com/a/58023653
Gatsby: https://css-tricks.com/how-to-the-get-current-page-url-in-gatsby/#method-2-get-the-href-property-of-location-data-from-props

Reply:
I changed it to check if window is defined before setting the value. If undefined, the value defaults to a warning about requiring the analytics function to handle window info.